src/components/ngRedux.js

```diff
@@ -1,10 +1,41 @@
 import Connector from './connector';
+import invariant from 'invariant';
+import isFunction from '../utils/isFunction';
+import {createStore, applyMiddleware} from 'redux';
 
 export default function ngReduxProvider() {
-  let reduxStore = undefined;
-  this.setReduxStore = store => reduxStore = store;
+  let _reducer = undefined;
+  let _middlewares = [];
+  let _storeEnhancer = undefined;
 
-  this.$get = () => {
-    return Connector(reduxStore);
+  this.createStoreWith = (reducer, middlewares, storeEnhancer) => {
+  	  invariant(
+        isFunction(reducer),
+        'The reducer parameter passed to createStoreWith must be a Function. Instead received %s.',
+        typeof reducer
+      );
+
+      invariant(
+        !storeEnhancer || isFunction(storeEnhancer),
+        'The storeEnhancer parameter passed to createStoreWith must be a Function. Instead received %s.',
+        typeof storeEnhancer
+      );
+
+      _reducer = reducer;
+      _storeEnhancer = storeEnhancer || createStore;
+      _middlewares = middlewares;
+  };
+
+  this.$get = ($injector) => {
+  	let resolvedMiddleware = [];
+  	for(let middleware of _middlewares) {
+  		if(_.isString(middleware)) {
```

Comment:
`_` is undefined 

but when changing it to `typeof middleware === string`, I was able to get it working.

Made a simple log middleware that used angular $log instead, and was able to get things working.

```javascript
.factory('logger', ['$log', function ($log) {
    return store => next => action => {
      $log.log('dispatching', action);
      let result = next(action);
      $log.log('next state', store.getState());
      return result;
    };

  }])
  .config(['$ngReduxProvider', ($ngReduxProvider) => {
    //let createStoreWithMiddleware = applyMiddleware(apiMiddleware, logger, thunk)(createStore);
    //let store = createStoreWithMiddleware(reducers);

    //$ngReduxProvider.setReduxStore(store);
    $ngReduxProvider.createStoreWith(reducers, [apiMiddleware, 'logger', thunk]);
  }])
```